sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3.java

```diff
@@ -247,9 +378,45 @@ public String getNamespace() {
       return namespace;
     }
 
+    /**
+     * The composite {@link PTransform} for reading from datastore.
+     *
+     * <p>It involves the following steps,
+     * <ul>
+     *   <li>Create a singleton of the user provided {@code query} and apply a {@link ParDo} that
+     *   splits the query into {@code numQuerySplits} and assign each split query a unique
+     *   {@code Integer} as the key. The resulting output is of the type
+     *   {@code PCollection<KV<Integer, Query>>}.
+     *
+     *   <p>If the value of {@code numQuerySplits} is less than or equal to 0, then the number of
+     *   splits will be computed dynamically based on the size of the data for the {@code query}.
+     *
+     *   <li>The resulting {@code PCollection} is sharded using a {@link GroupByKey} operation. The
+     *   queries are extracted from they {@code KV<Integer, Iterable<Query>>} and flattened to
+     *   output a {@code PCollection<Query>}.
+     *
+     *   <li>In the third step, a {@code ParDo} reads entities for each query and outputs
+     *   a {@code PCollection<Entity>}.
+     * </ul>
+     */
```

Comment:
`@inheritDoc`? Or, maybe convert from javadoc to just a `//`-prefix block. (Is this really for dev readers rather than pipeline authors?)